### ISSUE TITLE
SafetyNet fails after phone is running for a while

### ISSUE THREAD

**Issue by nickmykkanen** [topjohnwu/Magisk#585]
Device Info:
OnePlus 6 (A6003)
Oxygen OS 5.1.11 / Android 8.1.0
Security Patch Level July 1, 2018
Magisk 17.1
Magisk Manager 5.9.1
Additional device info can be provided if necessary - just let me know what you need.

SafetyNet is passing right away after boot (both CTS Profile and Basic Integrity). At some point later on, SafeyNet starts to fail. It seems to happen at random. Sometimes, it'll start failing within seconds of being booted. Other times, it'll pass just fine for a few hours, and then start failing. I have had absolutely no luck finding the culprit. I've disabled all battery optimization that I'm able to (in case there's a process being killed somewhere), have tried using core only mode, and have checked the logs for any indication of the issue, with no luck. After boot, SafetyNet passes, at which point I'll clear the logs, and check a few minutes later. SafetyNet will usually be failing by that point, and there is nothing in either logs.

This may be the same or similar issue that is occurring with #457 and/or #513. It's hard to say for sure.

There have been a few other people reporting the same issue on the Magisk subreddit. I haven't been able to find any solutions for the issue, and I don't believe anyone has found the cause yet.

**Comment by HeyItsJono** [comment]
I get this occasionally as well, but usually it's hours after boot, and a reboot fixes it. Not sure what logs to post.

**Comment by ahxxm** [comment]
I reported a [lowmemorykiller bug](https://github.com/topjohnwu/Magisk/issues/457#issuecomment-411684537) to upstream(lineage os developer), after it got fixed safetynet lasts longer then(but not forever), did you find anything unusual in adb log?

**Comment by silphide** [comment]
Happened to me as well. My workaround is re-toggle the Magisk Hide to pass the SafetyNet check for that moment.

**Comment by Chahk** [comment]
I noticed that this started happening when I added a work profile.  The work-around is to use the "Restore Magisk Manager with original package" option in Settings, then uninstall the Magisk Manager that appears in the Work Profile, and then Hide MM again.

**Comment by Cubox-** [comment]
I have this issue and never created nor used a work profile.

**Comment by jrtberlin** [comment]
I have the same issue and looked a bit into it.

## Specs 
- OnePlus 6t

- Buildnumber: ONEPLUS A6013_41_181103
- Android: 9 (November patch)
- OxygenOS: 9.0.5
- SoC: sdm845 (arm64-v8a)
- Magisk Version: v17.4-89e0be00

---

- I have no work profile
- I have only one module installed (F-Droid Priveleged Extension) and it works flawlessly.


## Observations

After a while SafetyNet tests fail.

- No consistency in the uptime after this happens. (I have seen uptimes around 11h without the issue)
![screenshot_20181114-083613](https://user-images.githubusercontent.com/14153182/48470402-9ce22100-e7f1-11e8-81dc-53276717ccd0.jpg)

- I have seen successful tests without a reboot after a failed test. (There were around 2 hours inbetween)

## Investigation

### Logs

<b>Could not find /factory/wv.keys. Falling Back to L3.</b>

```
11-13 14:21:25.572 E/DrmWidevineDash(718): hlos api version =  14
11-13 14:21:25.572 E/DrmWidevineDash(718): tz api version =  14
11-13 14:21:25.572 E/DrmWidevineDash(718): OEMCrypto_APIVersion: ends! returns version 14
11-13 14:21:25.573 E/DrmWidevineDash(718): hlos api version =  14
11-13 14:21:25.573 E/DrmWidevineDash(718): tz api version =  14
11-13 14:21:25.573 E/DrmWidevineDash(718): OEMCrypto_APIVersion: ends! returns version 14
11-13 14:21:25.573 E/DrmLibFs(569): Not in whitelist paths, not peappending
11-13 14:21:25.574 E/DrmLibFs(569): Not in whitelist paths, not peappending
11-13 14:21:25.575 E/DrmWidevineDash(718): Error: OEMCrypto_GetKeyData keyDataLength = 0
11-13 14:21:25.575 W/WVCdm   (718): [oemcrypto_adapter_dynamic.cpp(748):LoadLevel1] Could not find /factory/wv.keys. Falling Back to L3.
11-13 14:21:25.575 D/QSEECOMAPI(718): QSEECom_dealloc_memory 
11-13 14:21:25.575 D/QSEECOMAPI(718): QSEECom_shutdown_app, app_id = 174
11-13 14:21:25.577 D/WVCdm   (718): [crypto_session.cpp(646):Open] CryptoSession::Open: Lock: requested_security_level: Default
11-13 14:21:25.583 E/libc    (14078): Access denied finding property "vendor.debug.egl.swapinterval"
11-13 14:21:25.584 I/WVCdm   (718): [device_files.cpp(166):ExtractDeviceInfo] ExtractDeviceInfo Entry
11-13 14:21:25.588 I/WVCdm   (718): [cdm_engine.cpp(147):OpenSession] CdmEngine::OpenSession: sid97
11-13 14:21:25.588 I/WVCdm   (718): [cdm_engine.cpp(742):QueryOemCryptoSessionId] CdmEngine::QueryOemCryptoSessionId
11-13 14:21:25.588 I/WVCdm   (718): [cdm_engine.cpp(233):GenerateKeyRequest] CdmEngine::GenerateKeyRequest
11-13 14:21:25.589 D/WVCdm   (718): [license.cpp(355):PrepareKeyRequest] PrepareKeyRequest: nonce=3856258402
11-13 14:21:25.599 E/libc    (14078): Access denied finding property "vendor.debug.egl.swapinterval"
11-13 14:21:25.666 V/chatty  (14078): uid=10114(com.sp.SX4fpPZh.pQ4q) RenderThread identical 4 lines
11-13 14:21:25.682 E/libc    (14078): Access denied finding property "vendor.debug.egl.swapinterval"
11-13 14:21:25.695 D/FileSizeUtil(3650): getAvailableStoragePercentValue totalBytes=242142883840,freeBytes=223124148224,availableStoragePercent=0.9214565577381701
11-13 14:21:25.696 D/WidgetViewService(3650): availableStorage has no change,availableStorage:92%
11-13 14:21:25.699 E/libc    (14078): Access denied finding property "vendor.debug.egl.swapinterval"
11-13 14:21:25.715 E/libc    (14078): Access denied finding property "vendor.debug.egl.swapinterval"
11-13 14:21:25.731 I/WVCdm   (718): [cdm_engine.cpp(190):CloseSession] CdmEngine::CloseSession: sid97
11-13 14:21:25.732 E/libc    (14078): Access denied finding property "vendor.debug.egl.swapinterval"
11-13 14:21:25.739 I/WVCdm   (718): [(0):] L3 Terminate.
11-13 14:21:25.749 E/libc    (14078): Access denied finding property "vendor.debug.egl.swapinterval"

```

**Comment by lialosiu** [comment]
I have this problem too. I running magisk version `v17.4-a9f265a5`

A few days ago I thought it was MagiskHide not working.  At here -> #804 

But today, I found my game stop work again. And SafetyNet test was **fails**

Maybe my problem is same of this issue😂

**Comment by sfai05** [comment]
> Happened to me as well. My workaround is re-toggle the Magisk Hide to pass the SafetyNet check for that moment.

Same here. Toggle Magisk Hide do the tricks.
Using Pixel 3 with December Security Patch

**Comment by Cubox-** [comment]
Toggling magisk hide does not help in my case. Only a reboot can restore SafetyNet passing.

**Comment by Cubox-** [comment]
I commented previously thinking I had this issue, it is apparently using the application AdGuard making my SafetyNet fail.

**Comment by sachingopal** [comment]
I don't have any modules. A fresh install of Magisk 18 and safety check fails on the oneplus 6 with Android 9.0.2.

**Comment by sachingopal** [comment]
Ok, if I disable and enable magiskhide, the safetynet would pass. Not sure for how long. Kinda strange.

**Comment by Possum** [comment]
This is happening to me as well. It first started months ago, but until this past weekend relatively stable for me. Running stock OxygenOS 5.1.7 + Magisk V18.0 on OnePlus 5 (I just upgraded after it started failing on the previous OOS + Magisk hoping it would help). 

Like the other reports, toggling the Magisk Hide feature fixes it for a while, but that duration has become much shorter since this weekend. Only seems to work for a few hours now if I'm lucky.

**Comment by Dan-Wood** [comment]
@sfai05

How's you get around losing your Sim when switching magical hide on / off when it fails?

Or is my pixel 3 xl different to just the normal 3?

**Comment by sfai05** [comment]
> @sfai05
> 
> How'd you get around losing your Sim when switching magical hide on / off when it fails?
> 
> Or is my pixel 3 xl different to just the normal 3?

No. Didnt have any issue on my sim. Also the problem solved after I move back to Magisk 17.3 ( At least it work fine for few days already )

**Comment by phobiac** [comment]
I have experienced this error and found a way to reproduce it in my setup.

Magisk version is 18.0. I noticed that the presence of an offline Google Maps map immediately triggers a safetynet failure. Maps comes as a system app and this is a stock but rooted LGV20 ROM.

More details that may help include:
The offline map was downloaded to internal memory. 
I use the "location share" feature, which may be interacting with the offline map download in an unexpected way?
I do have Google Maps set to be hidden in Magiskhide.
